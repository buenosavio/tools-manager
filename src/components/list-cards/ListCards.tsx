import { useDispatch, useSelector } from 'react-redux'
import Card from '../card/Card'
import toolSelector from '../../store/tools/selector'
import { useEffect } from 'react'
import { Tag, Tool, toolActions } from '../../store/tools/duck'
import Loading from '../utils/Loading'
import { Text } from './ListCards.styles'
import { filteredItems } from '../../utils/filteredItems'

function ListCards({ callMe }: any) {
  const tools = useSelector(toolSelector.get)
  const isLoading = useSelector(toolSelector.isLoading)
  const search = useSelector(toolSelector.search)
  const isOnlyTag = useSelector(toolSelector.isOnlyTag)

  const dispatch = useDispatch()

  useEffect(() => {
    callMe(dispatch(toolActions.load()))
  }, [])

  const filtered: any = filteredItems(tools)

  return (
    <div>
      {!isLoading ? (
        tools.length > 0 ? (
          search.length > 0 ? (
            <>
              {filtered.map(({ item, i }: any) => (
                <Card item={item} key={i} role="alert" />
              ))}
            </>
          ) : (
            tools.map((tool, i) => <Card item={tool} key={i} role="alert" />)
          )
        ) : (
          <Text>No tools registered</Text>
        )
      ) : (
        <Loading />
      )}
    </div>
  )
}

export default ListCards
