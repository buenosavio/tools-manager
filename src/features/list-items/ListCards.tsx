import { useDispatch, useSelector } from 'react-redux'
import Card from './Card'
import toolSelector from '../../store/tools/selector'
import { useEffect } from 'react'
import { toolActions } from '../../store/tools/duck'
import Loading from '../utils/Loading'

function ListCards() {
  const tools = useSelector(toolSelector.get)
  const isLoading = useSelector(toolSelector.isLoading)

  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(toolActions.load())
  }, [])

  return (
    <div>
      {!isLoading ? (
        tools.length > 0 ? (
          tools.map((tool, i) => <Card item={tool} key={i} />)
        ) : (
          <p
            style={{
              display: 'flex',
              justifyContent: 'center',
              fontSize: '30px',
              marginTop: '30%'
            }}
          >
            No tools registered
          </p>
        )
      ) : (
        <Loading />
      )}
    </div>
  )
}

export default ListCards
