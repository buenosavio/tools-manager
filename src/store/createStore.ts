/* eslint-disable import/no-anonymous-default-export */
import { createStore, applyMiddleware, Reducer, Store } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { routerMiddleware } from 'connected-react-router';
import { SagaMiddleware } from 'redux-saga';
import { History } from 'history';

const isDevelopMode = process.env.NODE_ENV === 'development';

export default (reducers: Reducer, sagaMiddleware: SagaMiddleware, history: History): Store => {
  if (isDevelopMode) {
    const storeEnhacer = composeWithDevTools({ trace: true, traceLimit: 30 })(
      applyMiddleware(sagaMiddleware, routerMiddleware(history))
    );
    return createStore(reducers, storeEnhacer);
  }
  return createStore(reducers, applyMiddleware(sagaMiddleware, routerMiddleware(history)));
};
