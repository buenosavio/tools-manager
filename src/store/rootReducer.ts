import ToolsReducer from './tools/duck'
import {combineReducers} from '@reduxjs/toolkit'

const rootReducer = combineReducers({
  ToolsReducer
})


export type AppState = ReturnType<typeof rootReducer>

export default rootReducer