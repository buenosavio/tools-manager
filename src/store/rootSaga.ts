import {all} from 'redux-saga/effects'
import toolSaga from './tools/saga'

export default function* rootSaga(): Generator {
  yield all([toolSaga()])
}