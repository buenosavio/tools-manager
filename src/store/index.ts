import createSagaMiddleware from "redux-saga";
import createStore from "./createStore";
import rootReducer from "./rootReducer";
import history from "./history";
import { persistReducer } from "redux-persist";
import persistStore from "redux-persist/es/persistStore";
import rootSaga from "./rootSaga";
import storage from 'redux-persist/lib/storage'

const sagaMiddleware = createSagaMiddleware()

const store = createStore(persistReducer({key: 'store', storage },rootReducer), sagaMiddleware, history);

const persistor = persistStore(store);

sagaMiddleware.run(rootSaga)

export type AppDispatch = typeof store.dispatch;

export {store, persistor}