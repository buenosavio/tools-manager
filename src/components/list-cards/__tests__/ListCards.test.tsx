import ListCards from '../ListCards'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { store } from '../../../store'
import { mount, render } from 'enzyme'
import withProvider from '../../../utils/withContext'
import { screen } from '@testing-library/react'

describe('Test of component <ListCards />', () => {
  const Component = withProvider(ListCards, store)
  const mockCallMe = jest.fn()

  test('Snapshot of component should be ok', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ListCards />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('UseEffect to be called', () => {
    mount(<Component callMe={mockCallMe} />)
    expect(mockCallMe).toHaveBeenCalled()
  })
  test('Cards should be rendered on document', () => {})
})
