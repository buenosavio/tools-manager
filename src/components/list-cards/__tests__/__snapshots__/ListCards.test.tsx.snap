// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component <ListCards /> Snapshot of component should be ok 1`] = `
<div>
  <div
    className="sc-bjUoiL dEyjCG"
  >
    <img
      alt=""
      src="loading.gif"
    />
    <h2>
      Loading....
    </h2>
  </div>
</div>
`;
